src/PageUtils/PagePdf.php

```diff
@@ -11,10 +11,104 @@
 
 namespace HeadlessChromium\PageUtils;
 
+use HeadlessChromium\Communication\Message;
 use HeadlessChromium\Exception\PdfFailed;
+use HeadlessChromium\Page;
 
 class PagePdf extends AbstractBinaryInput
 {
+    private const TYPE_NUMERIC = 1;
+    private const TYPE_STRING = 2;
+    private const TYPE_BOOLEAN = 3;
+
+    /**
+     * @see https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-printToPDF
+     */
+    private const OPTIONS = [
+        'landscape' => self::TYPE_BOOLEAN,
+        'printBackground' => self::TYPE_BOOLEAN,
+        'displayHeaderFooter' => self::TYPE_BOOLEAN,
+        'headerTemplate' => self::TYPE_STRING,
+        'footerTemplate' => self::TYPE_STRING,
+        'paperWidth' => self::TYPE_NUMERIC,
+        'paperHeight' => self::TYPE_NUMERIC,
+        'marginTop' => self::TYPE_NUMERIC,
+        'marginBottom' => self::TYPE_NUMERIC,
+        'marginLeft' => self::TYPE_NUMERIC,
+        'marginRight' => self::TYPE_NUMERIC,
+        'pageRanges' => self::TYPE_STRING,
+        'ignoreInvalidPageRanges' => self::TYPE_BOOLEAN,
+        'preferCSSPageSize' => self::TYPE_BOOLEAN,
+        'scale' => self::TYPE_NUMERIC,
+    ];
+
+    /**
+     * @var Page
+     */
+    private $page;
+
+    private $options = [];
+
+    /**
+     * @throws \HeadlessChromium\Exception\CommunicationException
+     */
+    public function __construct(Page $page, array $options = [])
+    {
+        $this->page = $page;
+
+        $this->setOptions($options)->print();
+    }
+
+    /**
+     * @throws \HeadlessChromium\Exception\CommunicationException
+     */
+    public function print(): self
+    {
+        $responseReader = $this->page->getSession()->sendMessage(new Message('Page.printToPDF', $this->options));
+
+        parent::__construct($responseReader);
+
+        return $this;
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    public function setOptions(array $options): self
+    {
+        \array_map([$this, 'validateOption'], \array_keys($options), $options);
+
+        $this->options = $options;
+
+        return $this;
+    }
+
+    /**
+     * @param string                $name  Option name
```

Comment:
don't think the option name and value comments add any value. the fuction is called validateOption and has name and value params, so :trollface:

Reply:
It's to specify the types of `$value` since union types are not supported in php 7. It felt wrong to specify only one of the params. I can remove the descriptions and leave only the types if you think it's better.

Reply:
i was referring specifically to the text `Option name`, not the whole line. ;)

Reply:
Ok, I'll amend the commit later today when I get home.